src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/network/v4_9/CloudStackNetworkPlugin.java

```diff
@@ -80,73 +76,112 @@ public String requestInstance(NetworkOrder networkOrder, CloudStackUser cloudUse
                 .netmask(subnetInfo.getNetmask())
                 .build(this.cloudStackUrl);
 
-        CloudStackUrlUtil.sign(request.getUriBuilder(), cloudUser.getToken());
+        return doRequestInstance(request, cloudStackUser);
+    }
 
-        String jsonResponse = null;
-        try {
-            jsonResponse = this.client.doGetRequest(request.getUriBuilder().toString(), cloudUser);
-        } catch (HttpResponseException e) {
-            CloudStackHttpToFogbowExceptionMapper.map(e);
-        }
+    @Override
+    public NetworkInstance getInstance(@NotNull NetworkOrder networkOrder,
+                                       @NotNull CloudStackUser cloudStackUser)
+            throws FogbowException {
+
+        LOGGER.info(String.format(Messages.Info.GETTING_INSTANCE_S, networkOrder.getInstanceId()));
+        GetNetworkRequest getNetworkRequest = new GetNetworkRequest.Builder()
+                .id(networkOrder.getInstanceId())
+                .build(this.cloudStackUrl);
 
-        CreateNetworkResponse response = CreateNetworkResponse.fromJson(jsonResponse);
-        return response.getId();
+        return doGetInstance(getNetworkRequest, cloudStackUser);
     }
 
     @Override
-    public NetworkInstance getInstance(NetworkOrder networkOrder, CloudStackUser cloudUser) throws FogbowException {
-        GetNetworkRequest request = new GetNetworkRequest.Builder()
+    public void deleteInstance(@NotNull NetworkOrder networkOrder,
+                               @NotNull CloudStackUser cloudStackUser)
+            throws FogbowException {
+
+        LOGGER.info(String.format(Messages.Info.DELETING_INSTANCE_S, networkOrder.getInstanceId()));
+        DeleteNetworkRequest request = new DeleteNetworkRequest.Builder()
                 .id(networkOrder.getInstanceId())
                 .build(this.cloudStackUrl);
 
-        CloudStackUrlUtil.sign(request.getUriBuilder(), cloudUser.getToken());
+        doDeleteInstance(request, cloudStackUser);
+    }
+
+    @VisibleForTesting
+    void doDeleteInstance(@NotNull DeleteNetworkRequest request,
+                          @NotNull CloudStackUser cloudStackUser)
+            throws FogbowException {
+
+        URIBuilder uriRequest = request.getUriBuilder();
+        CloudStackUrlUtil.sign(uriRequest, cloudStackUser.getToken());
 
-        String jsonResponse = null;
         try {
-            jsonResponse = this.client.doGetRequest(request.getUriBuilder().toString(), cloudUser);
+            CloudStackCloudUtils.doGet(this.client, uriRequest.toString(), cloudStackUser);
```

Comment:
this method should be renamed. Besides you're just calling a url without any body, you're not doing a get request here, for example. You're doing a delete request. Maybe doRequest() should be better. Also check that this method has been called in doDeleteInstance test

Reply:
I agree with you. I will fix this. This name might cause misunderstood.  I will change to doRequest.
Only to clarify, all operations in the Cloudstack are a GET HTTP operation, therefore It does not matter whether is a delete, update or create. The previous name was that because of this situation.

Reply:
done